Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1644,18 +1659,19 @@ def readinto(self, b):
 def get_current_splunk_time(splunk_service: client.Service):
     t = datetime.utcnow() - timedelta(days=3)
     time = t.strftime(SPLUNK_TIME_FORMAT)
-    kwargs_oneshot = {'count': 1, 'earliest_time': time}
+    kwargs_oneshot = {'count': 1, 'earliest_time': time, 'output_mode': OUTPUT_MODE_JSON, }
     searchquery_oneshot = '| gentimes start=-1 | eval clock = strftime(time(), "%Y-%m-%dT%H:%M:%S")' \
                           ' | sort 1 -_time | table clock'
 
     oneshotsearch_results = splunk_service.jobs.oneshot(searchquery_oneshot, **kwargs_oneshot)
 
-    reader = results.ResultsReader(oneshotsearch_results)
+    reader = results.JSONResultsReader(oneshotsearch_results)
     for item in reader:
-        if isinstance(item, results.Message):
-            return item.message["clock"]
+        if handle_message(item):
+            continue
         if isinstance(item, dict):
             return item["clock"]
```

Comment:
earlier we first check if the type is dict, then message. Let's keep the same logic here, unless there's a reason